File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
                                                        
        int original = x;
        x = Math.abs(x);
        
        long reversed = 0;
        while (x != 0) {
            reversed = reversed * 10 + (x % 10);
            x /= 10;
        }
        
        if (original < 0) {
            reversed = -reversed;
        }
        
        if (reversed > Integer.MAX_VALUE || reversed < Integer.MIN_VALUE) {
            return 0;
        }
        
        return (int) reversed;
    }
}

File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        if x < 0:
            sign = -1
        else:
            sign = 1

        x = abs(x)
        
        rev = int(str(x)[::-1])
        rev = rev * sign
        
        if rev < -2**31 or rev > 2**31 - 1:
            return 0

        return rev

